src/main/java/com/sun/controller/RequestController.java

```diff
@@ -0,0 +1,98 @@
+package com.sun.controller;
+
+
+import com.sun.common.CurrentUser;
+import com.sun.entity.Book;
+import com.sun.service.BookService;
+import com.sun.service.RequestService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.servlet.http.HttpSession;
+import java.util.ArrayList;
+import java.util.List;
+
+@Controller
+@PreAuthorize("hasRole('ROLE_USER')")
+public class RequestController {
+    @Autowired
+    private RequestService requestService;
+    @Autowired
+    private BookService bookService;
+
+    @RequestMapping(value = "/add/{id}", method = RequestMethod.GET)
+    public String request(ModelMap model, HttpSession session, @PathVariable("id") String id) {
```

Comment:
```suggestion
    public String request(ModelMap model, HttpSession session, @PathVariable("id") String bookId) {
```
Nên ri em nhé, chứ Id chị bjk nó là gì, là id của request hay j ...